webviz_config/_oauth2.py

```diff
@@ -0,0 +1,162 @@
+import os
+import datetime
+from typing import Tuple
+
+import msal
+import flask
+
+
+class Oauth2:
+    """Oauth2 authorization Code grant flow"""
+
+    def __init__(self, app: flask.app.Flask):
+        self._app = app
+
+        # Azure AD app registration info (currently the values are taken from environment variables)
+        self._tenant_id = os.environ["WEBVIZ_TENANT_ID"]
+        self._client_id = os.environ["WEBVIZ_CLIENT_ID"]
+        self._client_secret = os.environ["WEBVIZ_CLIENT_SECRET"]
+        self._scope = os.environ["WEBVIZ_SCOPE"]
+
+        # Initiate msal
+        self._msal_app = msal.ConfidentialClientApplication(
+            client_id=self._client_id,
+            client_credential=self._client_secret,
+            authority=f"https://login.microsoftonline.com/{self._tenant_id}",
+        )
+        self._accounts = self._msal_app.get_accounts()
+
+        # Initiate oauth2 endpoints
+        self.set_oauth2_endpoints()
+
+    def set_oauth2_endpoints(self) -> None:
+        """/login and /auth-return endpoints are added for Oauth2 authorization
+        code flow.
+
+        At the end of the flow, a session cookie containing a valid access token
+        and its expiration date is created. This flask session object can be
+        accessed from Webviz plugin.
+
+        To get the access token: flask.session.get("access_token")
+        To get the expiration date: flask.session.get("expiration_date")
+
+        An Azure AD application should be registered, and the following environment
+        variables should be set: WEBVIZ_TENANT_ID, WEBVIZ_CLIENT_ID,
+        WEBVIZ_CLIENT_SECRET, WEBVIZ_SCOPE.
+
+        WEBVIZ_SESSION_SECRET_KEY environment variable is also needed for signing the
+        session cookie.
```

Comment:
Replaced by autogenerated key, isn't it?

Reply:
Ah yes, this should be removed :)